workflows/preparegenome.nf

```diff
@@ -52,18 +56,39 @@ workflow {
     // Create a FAI genome index using samtools
     ch_fai = SAMTOOLS_FAIDX ( ch_fasta ).fai
 
+    FILTER_GTF ( params.gtf )
+
+    // iCount segment the raw annotation
+    RAW_ICOUNT_SEGMENT (
+        params.gtf,
+        ch_fai
+    )
 
-    // Create genome segmentation file using iCount
-    ICOUNT_SEGMENT (
+    // iCount segment the filtered annotation
+    FILTERED_ICOUNT_SEGMENT (
         FILTER_GTF.out.post_filtering_gtf,
         ch_fai
     )
 
+    RENAME (
+        FILTERED_ICOUNT_SEGMENT.out.regions,
+        "filtered.regions.gtf.gz"
+    )
+
+    // Resolve unannotated for iCount summary and other processes
+    RESOLVE_UNANNOTATED (
+        RAW_ICOUNT_SEGMENT.out.regions,      // filtered_segmentation
+        RENAME.out.renamed, // unfiltered_segmentation
```

Comment:
are the raw segment and the filtered in the correct order here - filtered should be first I think? And why not RAW_ICOUNT_SEGMENT.out.regions and FILTERED_ICOUNT_SEGMENT.out.regions?

Reply:
I think you're commenting on an old commit, as I replaced the `RENAME` module with a better solution. This was needed as both the `iCount segment` processes outputted `regions.gtf.gz` and there was an input name conflict.

Regarding the ordering, you are correct, I've got this the wrong way around, I'll push a fix.

Reply:
cool!